yak/CMakeLists.txt

```diff
@@ -1,133 +1,35 @@
-cmake_minimum_required(VERSION 3.5.0)
-project(yak)
+cmake_minimum_required(VERSION 3.10.0)
+project(yak VERSION 0.2.0 LANGUAGES CUDA CXX)
 
-# Default to C++14 for cross-compatibility
-if(NOT CMAKE_CXX_STANDARD)
+if(NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD_REQUIRED True)
 endif()
 
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra)
+if(NOT DEFINED CMAKE_CUDA_STANDARD)
+  set(CMAKE_CUDA_STANDARD 14)
+  set(CMAKE_CUDA_STANDARD_REQUIRED True)
 endif()
 
-MACRO(CUDA_COMPUTE_TARGET_FLAGS arch_bin arch_ptx cuda_nvcc_target_flags)
-    string(REGEX REPLACE "\\." "" ARCH_BIN_WITHOUT_DOTS "${${arch_bin}}")
-    string(REGEX REPLACE "\\." "" ARCH_PTX_WITHOUT_DOTS "${${arch_ptx}}")
-
-    set(cuda_computer_target_flags_temp "")
-
-    # Tell NVCC to add binaries for the specified GPUs
-    string(REGEX MATCHALL "[0-9()]+" ARCH_LIST "${ARCH_BIN_WITHOUT_DOTS}")
-    foreach(ARCH IN LISTS ARCH_LIST)
-        if (ARCH MATCHES "([0-9]+)\\(([0-9]+)\\)")
-            # User explicitly specified PTX for the concrete BIN
-            set(cuda_computer_target_flags_temp ${cuda_computer_target_flags_temp} -gencode;arch=compute_${CMAKE_MATCH_2},code=sm_${CMAKE_MATCH_1};)
-        else()
-            # User didn't explicitly specify PTX for the concrete BIN, we assume PTX=BIN
-            set(cuda_computer_target_flags_temp ${cuda_computer_target_flags_temp} -gencode;arch=compute_${ARCH},code=sm_${ARCH};)
-        endif()
-    endforeach()
-
-    # Tell NVCC to add PTX intermediate code for the specified architectures
-    string(REGEX MATCHALL "[0-9]+" ARCH_LIST "${ARCH_PTX_WITHOUT_DOTS}")
-    foreach(ARCH IN LISTS ARCH_LIST)
-        set(cuda_computer_target_flags_temp ${cuda_computer_target_flags_temp} -gencode;arch=compute_${ARCH},code=compute_${ARCH};)
-    endforeach()
-
-    set(${cuda_nvcc_target_flags} ${cuda_computer_target_flags_temp})
-ENDMACRO()
-
-MACRO(APPEND_TARGET_ARCH_FLAGS)
-    set(cuda_nvcc_target_flags "")
-    CUDA_COMPUTE_TARGET_FLAGS(CUDA_ARCH_BIN CUDA_ARCH_PTX cuda_nvcc_target_flags)
-    if (cuda_nvcc_target_flags)
-        message(STATUS "CUDA NVCC target flags: ${cuda_nvcc_target_flags}")
-        list(APPEND CUDA_NVCC_FLAGS ${cuda_nvcc_target_flags})
-    endif()
-ENDMACRO()
-
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Modules/")
-message(STATUS "CMAKE MODULE PATH IS ${CMAKE_MODULE_PATH}")
-
-# TODO: Use built-in CUDA language support from CMake 3.8+ (requires different flags)
-#enable_language(CUDA)
+find_package(CUDA 9.0 REQUIRED)
+find_package(OpenCV 3 REQUIRED COMPONENTS core highgui)
+find_package(Eigen3 REQUIRED)
+find_package(PCL 1.8 REQUIRED COMPONENTS common io geometry)
```

Comment:
Recommend creating an alias target for PCL. Also on the libraries you should add PCL_DEFINITIONS.

Reply:
I added PCL_DEFINITIONS to the relevant libraries. I'll create a PCL alias target in a future PR.